File: algorithms/2_medium/D/02126_destroying_asteroids/solution.cpp
class Solution {
public:
    static bool asteroidsDestroyed(int mass, vector<int>& asteroids) {
        sort(asteroids.begin(), asteroids.end());
        int64_t curr_mass = mass;

        for (const auto a : asteroids) {
            if (a > curr_mass) {
                return false;
            }

            curr_mass += a;
        }

        return true;
    }
};

File: algorithms/2_medium/D/02126_destroying_asteroids/solution.rs
impl Solution {
    pub fn asteroids_destroyed(mass: i32, mut asteroids: Vec<i32>) -> bool {
        asteroids.sort_unstable();
        let mut curr_mass = mass as i64;

        for a in asteroids.into_iter() {
            if a as i64 > curr_mass {
                return false;
            }

            curr_mass += a as i64;
        }

        true
    }
}
